### PR TITLE
[POAE7-2366] between...and... string support(cast str to date tpch-q7 workaround)

### PR DESCRIPTION
**Pull request by kd97**
### What changes were proposed in this pull request?

between...and... str support
### Why are the changes needed?

cast str to date workaround to fix tpch q7.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

UTs
### Which label does this PR belong to?

BUG

### CODE REVIEW

File: cider/tests/functionality/CiderStringTest.cpp [hunk 1]
```diff
@@ -86,6 +86,10 @@ class CiderStringToDateTest : public CiderTestBase {
 };
 
 TEST_F(CiderStringToDateTest, DateStrTest) {
+  assertQuery(
+      "select col_str from test where col_str between date '1970-01-01' and date "
+      "'2077-12-31'",
+      "cast_str_to_date_implictly.json");
```
Comment (yma11) [inline cider/tests/functionality/CiderStringTest.cpp:92]: How about adding a test with null covered?

Comment (kd97) [reply cider/tests/functionality/CiderStringTest.cpp:92]: class `DateStrTest `will generate  varchar type null value with a 50% chance

https://github.com/intel/BDTK/blob/d46dfd8703ab19b023b1c52d3096289cd24bad8a/cider/tests/functionality/CiderStringTest.cpp#L76-L85

**Review by yma11 — APPROVED** [review]

**Review by winningsix — APPROVED** [review]
